crates/gateway-types/src/reply.rs

```diff
@@ -29,9 +29,60 @@ pub struct Block {
     pub timestamp: StarknetBlockTimestamp,
     pub transaction_receipts: Vec<transaction::Receipt>,
     pub transactions: Vec<transaction::Transaction>,
-    /// Version metadata introduced in 0.9.1, older blocks will not have it.
-    #[serde(default)]
-    pub starknet_version: Option<String>,
+    #[serde_as(as = "AlmostSemver")]
+    #[serde(default = "ancient_version")]
+    pub starknet_version: StarknetVersion,
```

Comment:
Don't we lose some data due to the truncating below?

Reply:
We lose the 4 segment yes. I thought it wasn't worth saving; but we could just continue storing the raw string and add a method to retrieve the semver version? That actually sounds pretty good to me now.

Reply:
Yeah, I'd definitely prefer keeping the original version string in the DB.